ipa-core/src/protocol/prss/mod.rs

```diff
@@ -133,6 +104,16 @@ impl From<RecordId> for PrssIndex {
     }
 }
 
+impl AddAssign<u32> for PrssIndex {
+    fn add_assign(&mut self, rhs: u32) {
+        if let Some(v) = self.0.checked_add(rhs) {
+            self.0 = v;
```

Comment:
This would be an end run around our limits.  If we cap the offset, then we should also cap the value here.  In this case, we would be ensuring that `rhs < (1<<11) && (self.0 + rhs) & 0xfffff_f800 == 0`.

Reply:
If we enforce it at `PrssIndex128`, then it seems we could keep the check in one place?